devito/data/decomposition.py

```diff
@@ -1,7 +1,7 @@
 from collections.abc import Iterable
 
 import numpy as np
-from cached_property import cached_property
+from functools import cached_property
```

Comment:
This is now a system import so convention throughout is to group it with system import (first group with collection)